events/apigw.go

```diff
@@ -42,6 +42,38 @@ type APIGatewayProxyRequestContext struct {
 	APIID         string                    `json:"apiId"` // The API Gateway rest API Id
 }
 
+// APIGatewayV2Request contains data coming from the new HTTP API Gateway
+type APIGatewayV2Request struct {
+	Version               string                     `json:"version"`
+	RouteKey              string                     `json:"routeKey"`
+	RawPath               string                     `json:"rawPath"`
+	RawQueryString        string                     `json:"rawQueryString"`
+	Cookies               []string                   `json:"cookies"`
+	Headers               map[string]string          `json:"headers"`
+	QueryStringParameters map[string]string          `json:"queryStringParameters"`
+	PathParameters        map[string]string          `json:"pathParameters"`
+	RequestContext        APIGatewayV2RequestContext `json:"requestContext"`
+	Body                  string                     `json:"body"`
+	IsBase64Encoded       bool                       `json:"isBase64Encoded,omitempty"`
+}
+
+// APIGatewayV2RequestContext contains the information to identify the AWS account and resources invoking the
+// Lambda function. It also includes Cognito identity information for the caller.
+type APIGatewayV2RequestContext struct {
+	RouteID      string                    `json:"routeId"`
+	RouteKey     string                    `json:"routeKey"`
+	AccountID    string                    `json:"accountId"`
+	Stage        string                    `json:"stage"`
+	RequestID    string                    `json:"requestId"`
+	Identity     APIGatewayRequestIdentity `json:"identity"`
+	Authorizer   map[string]interface{}    `json:"authorizer"`
```

Comment:
It looks like this can be better typed than `interface{}`. [The sample data](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) shows what the optional jwt integration looks like.

something like:
```
type APIGatewayV2JWTAuthorizier struct {
  Claims map[string]string `json:claims`
  Scopes []string          `json:scopes`
}
type APIGatewayV2RequestContext struct {
  ...
  Authorizer map[string]*APIGatewayV2JWTAuthorizer `json:authorizer`
  ...
}
```

Reply:
I took it from the APIGatewayProxyRequestContext which is from V1. But you are right, sample data shows JWT integration so we can define it as struct instead of using interface.